saged/disease_vector.py

```diff
@@ -0,0 +1,170 @@
+    batch_size = simulation_config['batch_size']
+    epochs = simulation_config['epochs']
+    kappa = simulation_config['kappa']
+    intermediate_dim = simulation_config['intermediate_dim']
+    latent_dim = simulation_config['latent_dim']
+    epsilon_std = simulation_config['epsilon_std']
+    val_frac = simulation_config['val_frac']
+
+    # Set tf to be deterministic
+    session_conf = tf.ConfigProto(
+        intra_op_parallelism_threads=1, inter_op_parallelism_threads=1
+    )
+
+    # Create tf session
+    sess = tf.Session(graph=tf.get_default_graph(), config=session_conf)
+    K.set_session(sess)
+
+    # Train a VAE on the data
+    expression_array, _ = expression_data.get_all_data()
+    expression_array = labeled_scaler.transform(expression_array)
+
+    sample_ids = expression_data.get_samples()
+
+    metadata['sample_ids'] = sample_ids
+
+    expression_df = pd.DataFrame(expression_array, index=sample_ids)
+
+    sample_to_label = expression_data.sample_to_label
+
+    samples = expression_df.index
+    labels = samples.map(sample_to_label)
+
+    disease_df = expression_df[labels.str.match(args.label)]
+    healthy_df = expression_df[labels.str.match(args.negative_class)]
+
+    expression_encoder, expression_decoder, hist = vae.run_tybalt_training(expression_df,
+                                                                           learning_rate,
+                                                                           batch_size,
+                                                                           epochs,
+                                                                           kappa,
+                                                                           intermediate_dim,
+                                                                           latent_dim,
+                                                                           epsilon_std,
+                                                                           val_frac,)
+
+    # Calculate the centroids for the embeddings of the disease and healthy samples
+    disease_embeddings = expression_encoder.predict_on_batch(disease_df)
+    healthy_embeddings = expression_encoder.predict_on_batch(healthy_df)
+
+    disease_centroid = disease_embeddings.mean(axis=0)
+    healthy_centroid = healthy_embeddings.mean(axis=0)
+    disease_std = disease_embeddings.std(axis=0)
+    healthy_std = healthy_embeddings.std(axis=0)
+
+    # Sample disease data
+    full_sample = np.zeros((args.sample_count, disease_embeddings.shape[1]))
+    for i in range(args.sample_count):
+        full_sample[i, :] = np.random.normal(loc=disease_centroid,
+                                             scale=disease_std)
+
+    sim_disease = expression_decoder.predict_on_batch(full_sample)
+
+    out_path = os.path.join(args.out_dir, 'joint_sim_disease.pkl')
+    with open(out_path, 'wb') as out_file:
+        pickle.dump(sim_disease.T, out_file)
+
+    # TODO save result
```

Comment:
Looks like you're doing this already, so maybe you can remove this TODO?

Reply:
Good catch